src/main/java/software/amazon/qldb/QldbDriverImpl.java

```diff
@@ -134,12 +141,18 @@ public void execute(ExecutorNoReturn executor, RetryPolicy retryPolicy) {
         }
 
         ExecutionContext executionContext = new ExecutionContext();
+        int getSessionAttempt = 0;
         while (true) {
             QldbSession qldbSession = null;
             try {
+                getSessionAttempt += 1;
```

Comment:
This variable is named poorly. qldbSession.execute could be the source of the ISE, not only getSession.

Reply:
Renaming to iseRetryAttempt and moving the increment down.